Makefile

```diff
@@ -0,0 +1,154 @@
+
+PRINT = echo
+
+ifeq ($(V),)
+	Q=@
+	MAKE += -s
+	CARGO += -q
+define out_console
+	$(PRINT) -e "[$(1)]\t$(2)"
+endef
+
+.SILENT:
+endif
+
+ifeq ($(NOVENDOR),)
+    # This MUST be kept in sync with API_HEADERS under lib.rs in libbpf-sys
+    LIBBPF_API_HEADERS := bpf.h \
+                          libbpf.h \
+                          btf.h \
+                          bpf_helpers.h \
+                          bpf_helper_defs.h \
+                          bpf_tracing.h \
+                          bpf_endian.h \
+                          bpf_core_read.h \
+                          libbpf_common.h \
+                          usdt.bpf.h
+
+    LIBBPF_SYS_LIBBPF_BASE_PATH := $(dir $(shell cargo metadata --format-version=1 | jq -r '.packages | .[] | select(.name == "libbpf-sys") | .manifest_path'))
+    LIBBPF_SYS_LIBBPF_INCLUDES :=  $(wildcard $(addprefix $(LIBBPF_SYS_LIBBPF_BASE_PATH)/libbpf/src/, $(LIBBPF_API_HEADERS)))
+    LIBBPF_INCLUDES := $(ROOT_DIR)/src/.out
+    LIBBPF_INCLUDE_FILES := $(addprefix $(LIBBPF_INCLUDES)/bpf/,$(LIBBPF_API_HEADERS))
+endif
+
+FILTER_INCLUDES := src/core/filters/packets/bpf/include \
+                   src/core/filters/meta/bpf/include
+INCLUDES_ALL := $(abspath $(wildcard src/core/probe/bpf/include \
+                                     src/core/probe/kernel/bpf/include \
+                                     src/core/probe/user/bpf/include \
+                                     src/core/events/bpf/include \
+                                     src/core/tracking/bpf/include \
+                                     /usr/include/x86_64-linux-gnu \
+                                     vendor/linux/include \
+                                     vendor/linux/asm/x86/include \
+                                     $(FILTER_INCLUDES)))
+INCLUDES_ALL += $(LIBBPF_INCLUDES)
+
+OVS_INCLUDES := $(abspath src/module/ovs/bpf/include)
+INCLUDES := $(addprefix -I, $(INCLUDES_ALL))
+
+EBPF_PROBES := $(abspath src/core/probe/kernel/bpf \
+                        src/core/probe/user/bpf)
+
+GENERIC_HOOKS := $(abspath src/module/skb/bpf \
+                           src/module/skb_drop/bpf \
+                           src/module/skb_tracking/bpf \
+                           src/module/nft/bpf \
+                           src/module/ct/bpf)
+
+OVS_HOOKS := $(abspath src/module/ovs/bpf)
+OUT_NAME := HOOK
+
+JOBS := $(patsubst -j%,%,$(filter -j%,$(MAKEFLAGS)))
+
+ifneq ($(JOBS),)
+    CARGO_JOBS := $(JOBS)
+else
+    CARGO_JOBS := 1
+endif
+
+all: build
+
+install: build
+	$(CARGO) $(CARGO_COMMON_OPTS) install $(CARGO_CMD_OPTS)
+
+release: CARGO_CMD_OPTS += --release
+release: build
+
+build: ebpf
+	$(call out_console,CARGO,building retis ($(if $(findstring --release,$(CARGO_CMD_OPTS)),release,debug))...)
+	$(Q)CARGO_BUILD_JOBS=$(CARGO_JOBS) \
+	RETIS_PKG_VERSION=$(RELEASE_VERSION) \
```

Comment:
Nit: s/RETIS_PKG_VERSION/RETIS_RELEASE_VERSION/

I think the name/version should be set for all targets. But we should add logic (could be done later) to detect if HEAD points to the right version tag otherwise add `+` at the end of the version. Same, if uncommited changes are detected we should add `-dirty`.

Reply:
that's right. 
about `HEAD`, I agree, I would not leverage this PR for this. It's a good follow-up, though.